bundles/org.openhab.core.io.websocket/src/main/java/org/openhab/core/io/websocket/package-info.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+@org.osgi.annotation.bundle.Header(name = org.osgi.framework.Constants.DYNAMICIMPORT_PACKAGE, value = "*")
```

Comment:
Is this dynamic import really necessary?

Reply:
Yes, even though I don't fully understand why. But I found that this is the solution for https://github.com/openhab/openhab-core/pull/3252#issuecomment-1374418213.

Reply:
Did you also try adding an import for `org.eclipse.jetty.websocket.server.WebSocketServerFactory` in the code instead? That way bnd will add an import for the package to the manifest.

Reply:
That works, too. Why is that superior to the dynamic-import?

Reply:
It will make sure that the required package imports are satisfied before the bundle is started so you'll know well in advance that something is wrong. That can be long before the code is executed that looks up the class.